### PR TITLE
Add krona task to TheiaMeta_Illumina_PE

### PR DESCRIPTION
**Pull request by cimendes**
Closes <Issue number>

## :hammer_and_wrench: Changes Being Made

This PR adds new functionality to TheiaMeta: **[Krona Charts](https://github.com/marbl/Krona/wiki)**

Krona allows hierarchical data to be explored with zooming, multi-layered pie charts. 
KronaChart

On Terra, **an HTML file** is populated on the datatable that can be downloaded and explored on the browser interactively for both raw and read screen taxonomic assessment.  

## :brain: Context and Rationale

Krona allows for the interactive visualization of the entire community at a high level. It also allows for to drill down for taxa of interest, keeping the taxonomic relationships. 

## :clipboard: Workflow/Task Steps

Viral Metagenomic workflow - TheiaMeta_Illumina_PE

### Inputs

N/A

### Outputs

4 new outputs have been added to TheiaMeta:
```
    String krona_version = krona_raw.krona_version
    String krona_docker = krona_raw.krona_docker
    File krona_html_raw = krona_raw.krona_html
    File krona_html_clean = krona_clean.krona_html
```

### Locally

Tested locally with `miniwdl`, both at the task level and the entire TheiaMeta workflow

```
2023-10-11 09:04:27.829 wdl.w:theiameta_illumina_pe done
{
  "dir": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe",
  "outputs": {
    "theiameta_illumina_pe.assembly_fasta": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/assembly_fasta/HAV0024.fasta",
    "theiameta_illumina_pe.assembly_length": 7499,
    "theiameta_illumina_pe.assembly_mean_coverage": 2436.59,
    "theiameta_illumina_pe.average_read_length": 157.0,
    "theiameta_illumina_pe.bbduk_docker": "us-docker.pkg.dev/general-theiagen/staphb/bbtools:38.76",
    "theiameta_illumina_pe.bedtools_docker": "us-docker.pkg.dev/general-theiagen/staphb/bedtools:2.31.0",
    "theiameta_illumina_pe.bedtools_version": "v2.31.0",
    "theiameta_illumina_pe.contig_number": 1,
    "theiameta_illumina_pe.fastq_scan_docker": "quay.io/biocontainers/fastq-scan:0.4.4--h7d875b9_1",
    "theiameta_illumina_pe.fastq_scan_version": "fastq-scan 0.4.4",
    "theiameta_illumina_pe.kraken2_docker": "us-docker.pkg.dev/general-theiagen/staphb/kraken2:2.1.2-no-db",
    "theiameta_illumina_pe.kraken2_percent_human_clean": 24.04,
    "theiameta_illumina_pe.kraken2_percent_human_raw": 68.11,
    "theiameta_illumina_pe.kraken2_report_clean": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/kraken2_report_clean/HAV0024.report.txt",
    "theiameta_illumina_pe.kraken2_report_raw": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/kraken2_report_raw/HAV0024.report.txt",
    "theiameta_illumina_pe.kraken2_version": "2.1.2",
    "theiameta_illumina_pe.krona_docker": "us-docker.pkg.dev/general-theiagen/biocontainers/krona:2.7.1--pl526_5", <---
    "theiameta_illumina_pe.krona_html_clean": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/krona_html_clean/HAV0024_krona.html",  <--- 
    "theiameta_illumina_pe.krona_html_raw": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/krona_html_raw/HAV0024_krona.html",  <---
    "theiameta_illumina_pe.krona_version": "2.7.1", <---
    "theiameta_illumina_pe.largest_contig": 7499, 
    "theiameta_illumina_pe.metaspades_docker": "us-docker.pkg.dev/general-theiagen/biocontainers/spades:3.12.0--h9ee0642_3",
    "theiameta_illumina_pe.metaspades_version": "v3.12.0",
    "theiameta_illumina_pe.minimap2_docker": "us-docker.pkg.dev/general-theiagen/staphb/minimap2:2.22",
    "theiameta_illumina_pe.minimap2_version": "2.22-r1101",
    "theiameta_illumina_pe.ncbi_scrub_docker": "us-docker.pkg.dev/general-theiagen/ncbi/sra-human-scrubber:1.0.2021-05-05",
    "theiameta_illumina_pe.num_reads_clean1": 246155,
    "theiameta_illumina_pe.num_reads_clean2": 246155,
    "theiameta_illumina_pe.num_reads_clean_pairs": "246155",
    "theiameta_illumina_pe.num_reads_raw1": 1607300,
    "theiameta_illumina_pe.num_reads_raw2": 1607300,
    "theiameta_illumina_pe.num_reads_raw_pairs": "1607300",
    "theiameta_illumina_pe.percent_coverage": 99.08,
    "theiameta_illumina_pe.percentage_mapped_reads": 21.0312,
    "theiameta_illumina_pe.pilon_docker": "us-docker.pkg.dev/general-theiagen/biocontainers/pilon:1.24--hdfd78af_0",
    "theiameta_illumina_pe.pilon_version": "1.24",
    "theiameta_illumina_pe.quast_docker": "us-docker.pkg.dev/general-theiagen/staphb/quast:5.0.2",
    "theiameta_illumina_pe.quast_version": "QUAST v5.0.2",
    "theiameta_illumina_pe.read1_clean": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read1_clean/HAV0024_1.clean.fastq.gz",
    "theiameta_illumina_pe.read1_dehosted": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read1_dehosted/HAV0024_R1_dehosted.fastq.gz",
    "theiameta_illumina_pe.read1_mapped": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read1_mapped/assembled_HAV0024_1.fq.gz",
    "theiameta_illumina_pe.read1_unmapped": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read1_unmapped/unassembled_HAV0024_1.fq.gz",
    "theiameta_illumina_pe.read2_clean": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read2_clean/HAV0024_2.clean.fastq.gz",
    "theiameta_illumina_pe.read2_dehosted": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read2_dehosted/HAV0024_R2_dehosted.fastq.gz",
    "theiameta_illumina_pe.read2_mapped": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read2_mapped/assembled_HAV0024_2.fq.gz",
    "theiameta_illumina_pe.read2_unmapped": "/home/ines_mendes/WDL/20231011_083247_theiameta_illumina_pe/out/read2_unmapped/unassembled_HAV0024_2.fq.gz",
    "theiameta_illumina_pe.samtools_docker": "us-docker.pkg.dev/general-theiagen/staphb/samtools:1.17",
    "theiameta_illumina_pe.samtools_version": "1.17",
    "theiameta_illumina_pe.theiameta_illumina_pe_analysis_date": "2023-10-11",
    "theiameta_illumina_pe.theiameta_illumina_pe_version": "PHB v1.2.0",
    "theiameta_illumina_pe.trimmomatic_docker": "us-docker.pkg.dev/general-theiagen/staphb/trimmomatic:0.39",
    "theiameta_illumina_pe.trimmomatic_version": "Trimmomatic 0.39"
  }
}
```

### Terra

~~Underway~~
21 blood samples containing DENV: https://app.terra.bio/#workspaces/theiagen-validations/Theiagen_Mendes_Sandbox/job_history/e59ebd75-006d-4c29-82be-bda2d4ed0b15

## :microscope: Quality checks

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The workflow/task has been tested locally and on Terra
- [x] The CI/CD has been adjusted and tests are passing
- [x] Everything follows the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

### CODE REVIEW

File: workflows/metagenomics/wf_theiameta_illumina_pe.wdl [hunk 1]
```diff
@@ -47,6 +53,11 @@ workflow theiameta_illumina_pe {
       classified_out = "classified#.fastq",
       unclassified_out = "unclassified#.fastq"
   }
+  call krona_task.krona as krona_clean {
+    input:
+      kraken2_report = kraken2_raw.kraken2_report,
```
Comment (sage-wright) [inline workflows/metagenomics/wf_theiameta_illumina_pe.wdl:58]: you might want to use the kraken2_clean here

Comment (cimendes) [reply workflows/metagenomics/wf_theiameta_illumina_pe.wdl:58]: This is a bug! Thank you so much for catching it! Do you have the availability to do a small fix @sage-wright?

**Comment by sage-wright** [comment]
Testing here: https://app.terra.bio/#workspaces/cdc-terra-resources/Theiagen_Wright_SC2_Sandbox/job_history/9b1b1626-dcbe-4586-9b16-14960c0ec81b

**Review by sage-wright — APPROVED** [review]
will squish once tests succeed!

**Comment by sage-wright** [comment]
Well done!

**Pull request merged**